### PR TITLE
backward-cpp: Use libdwarf 0.9.1

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
Specify library name and version:  **backward-cpp/all**

#22629 removes `libdwarf/20191104`, this PR updates to use the most recent version

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by RubenRBS — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by CJCombrink — APPROVED** [review]

**Comment by ericLemanissier** [comment]
Can this PR be retriggered by merging master into its branch, or by modifying the PR description, so that it can be merged ? thanks !

**Review by RubenRBS — DISMISSED** [review]

File: recipes/backward-cpp/all/conanfile.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def requirements(self):
             if self._has_stack_walking("libunwind"):
                 self.requires("libunwind/1.7.2", transitive_headers=True)
             if self._has_stack_details("dwarf"):
-                self.requires("libdwarf/20191104", transitive_headers=True, transitive_libs=True)
+                self.requires("libdwarf/0.9.1", transitive_headers=True, transitive_libs=True)
                 self.requires("libelf/0.8.13")
```
Comment (ericLemanissier) [inline recipes/backward-cpp/all/conanfile.py:85]:
```suggestion
                self.requires("libelf/0.8.13", transitive_headers=True)
```
https://github.com/bombela/backward-cpp/blob/v1.6/backward.hpp#L249

**Review by ericLemanissier — CHANGES_REQUESTED** [review]

File: recipes/backward-cpp/all/conanfile.py (hunk 1 above)
Comment (jeremy-rifkin) [reply recipes/backward-cpp/all/conanfile.py]: Thanks @ericLemanissier, I wasn't sure how to go about resolving :)

**Comment by jeremy-rifkin** [comment]
Hmm, will look tomorrow

**Comment by ericLemanissier** [comment]
I did some tests: the libelf inclusion problem is fixed by https://github.com/ericLemanissier/conan-center-index/commit/d9d1a5e90ccd8dd3d904fa85bff56b912221a650
However, after this, we fail with 
```
In file included from /home/codespace/.conan2/p/b/backw0b6e5828cc090/b/src/backward.cpp:26:
/home/codespace/.conan2/p/b/backw0b6e5828cc090/b/src/backward.hpp: In static member function ‘static int backward::TraceResolverLinuxImpl<backward::trace_resolver_tag::libdwarf>::close_dwarf(Dwarf_Debug)’:
/home/codespace/.conan2/p/b/backw0b6e5828cc090/b/src/backward.hpp:1836:34: error: too many arguments to function ‘int dwarf_finish(Dwarf_Debug)’
 1836 |   return dwarf_finish(dwarf, NULL);
      |                                  ^
In file included from /home/codespace/.conan2/p/b/backw0b6e5828cc090/b/src/backward.hpp:224,
                 from /home/codespace/.conan2/p/b/backw0b6e5828cc090/b/src/backward.cpp:26:
/home/codespace/.conan2/p/b/libdw43d67277842f3/p/include/libdwarf.h:1640:12: note: declared here
 1640 | DW_API int dwarf_finish(Dwarf_Debug dw_dbg);
      |            ^~~~~~~~~~~~
```
This is because backward-cpp does not support recent libdwarf : https://github.com/bombela/backward-cpp/issues/298 https://github.com/bombela/backward-cpp/issues/232

The issue above recommends switching to bfd, unfortunately binutils recipe only supports linux, and somehow does not expose bfd library and headers https://github.com/conan-io/conan-center-index/blob/master/recipes/binutils/all/conanfile.py#L184.

I'm not sure what's the best course of action here

**Comment by jeremy-rifkin** [comment]
That's unfortunate... I guess either backward-cpp needs to update or the part of https://github.com/conan-io/conan-center-index/pull/22629 removing `libdwarf/20191104` should be reverted.

**Comment by AbrilRBS** [comment]
We can keep this as-is, the binary will remain in conancenter even if not present in the git repo

**Pull request closed without merging**